soroban-built-in-token-spec/tests/test.rs

```diff
@@ -0,0 +1,96 @@
+#![cfg(feature = "testutils")]
+
+use soroban_auth::{
+    testutils::ed25519::{generate, sign},
+    Signature,
+};
+use soroban_sdk::{contractimpl, contracttype, symbol, BigInt, BytesN, Env, IntoVal};
+
+mod token_contract;
+use token_contract::{Identifier, TokenClient, TokenMetadata};
+
+#[contracttype]
+pub enum DataKey {
+    Token,
+}
+
+fn get_contract_id(e: &Env) -> Identifier {
+    Identifier::Contract(e.get_current_contract().into())
+}
+
+fn get_token(e: &Env) -> BytesN<32> {
+    e.contract_data().get_unchecked(DataKey::Token).unwrap()
+}
+
+pub struct TestContract;
+
+#[contractimpl]
+impl TestContract {
+    pub fn init(e: Env, salt: BytesN<32>) {
+        let id = e.deployer().from_current_contract(salt).deploy_token();
+        let metadata = TokenMetadata {
+            name: "name".into_val(&e),
+            symbol: "symbol".into_val(&e),
+            decimals: 7u32,
+        };
+        TokenClient::new(&e, &id).init_token(&get_contract_id(&e), &metadata);
+
+        e.contract_data().set(DataKey::Token, id);
+    }
+
+    pub fn get_token(e: Env) -> BytesN<32> {
+        get_token(&e)
+    }
+
+    pub fn mint(e: Env, to: Identifier, amount: BigInt) {
+        TokenClient::new(&e, get_token(&e)).mint(
+            &Signature::Invoker,
+            &BigInt::zero(&e),
+            &to,
+            &amount,
+        );
+    }
+
+    pub fn set_admin(e: Env, new_admin: Identifier) {
+        TokenClient::new(&e, get_token(&e)).set_admin(
+            &Signature::Invoker,
+            &BigInt::zero(&e),
+            &new_admin,
+        );
+    }
+}
+
+#[test]
+fn test() {
```

Comment:
💯 Great idea to write a test that uses the native token contract with the interface generated.